### PR TITLE
Upgrade storybook

### PR DESCRIPTION
**Pull request by TheSonOfThomp**
## ✍️ Proposed changes

- Upgrades all `@storybook` packages to latest
- Adds custom stories logic to prevent nested stories from being included in storybook build
- Fixes TS errors in main.ts

## 🧪 How to test changes

```bash
> yarn build-storybook
```

### CODE REVIEW

**Comment by bruugey** [comment]
Seeing some errors in the terminal
```

**Comment by bruugey** [comment]
```
@leafygreen-ui/confirmation-modal:tsc: ../../node_modules/@storybook/react/node_modules/type-fest/source/camel-case.d.ts(25,32): error TS1005: '?' expected.
@leafygreen-ui/confirmation-modal:tsc: ../../node_modules/@storybook/react/node_modules/type-fest/source/camel-case.d.ts(26,42): error TS1005: '?' expected.
@leafygreen-ui/confirmation-modal:tsc: ../../node_modules/@storybook/react/node_modules/type-fest/source/join.d.ts(59,36): error TS1005: '?' expected.
@leafygreen-ui/confirmation-modal:tsc: ../../node_modules/@storybook/react/node_modules/type-fest/source/join.d.ts(60,49): error TS1005: '?' expected.
@leafygreen-ui/confirmation-modal:tsc: ../../node_modules/@storybook/react/node_modules/type-fest/source/join.d.ts(64,50): error TS1005: '?' expected.
@leafygreen-ui/confirmation-modal:tsc: ../../node_modules/@storybook/react/node_modules/type-fest/source/join.d.ts(65,36): error TS1005: '?' expected.
```

**Comment by TheSonOfThomp** [comment]
> ```
> @leafygreen-ui/confirmation-modal:tsc: ../../node_modules/@storybook/react/node_modules/type-fest/source/camel-case.d.ts(25,32): error TS1005: '?' expected.
> @leafygreen-ui/confirmation-modal:tsc: ../../node_modules/@storybook/react/node_modules/type-fest/source/camel-case.d.ts(26,42): error TS1005: '?' expected.
> @leafygreen-ui/confirmation-modal:tsc: ../../node_modules/@storybook/react/node_modules/type-fest/source/join.d.ts(59,36): error TS1005: '?' expected.
> @leafygreen-ui/confirmation-modal:tsc: ../../node_modules/@storybook/react/node_modules/type-fest/source/join.d.ts(60,49): error TS1005: '?' expected.
> @leafygreen-ui/confirmation-modal:tsc: ../../node_modules/@storybook/react/node_modules/type-fest/source/join.d.ts(64,50): error TS1005: '?' expected.
> @leafygreen-ui/confirmation-modal:tsc: ../../node_modules/@storybook/react/node_modules/type-fest/source/join.d.ts(65,36): error TS1005: '?' expected.
> ```

This is an issue with Storybook's dependency `type-fest`. Looking to see if we can resolve this on our end

File: .storybook/preview.ts [hunk 1]
```diff
@@ -36,13 +36,18 @@ const parameters = {
...
-        'Sample Pages (WIP)',
+        'Components',
         'Contexts',
-        'Packages',
+        'Hooks',
+        'LeafyGreen Provider',
+        'Demo',
+        'Form',
+        'Lib Utilities',
+        'Sample Pages (WIP)',
```
Comment (spark33) [inline .storybook/preview.ts:50]: It feels like these should be repo specific - `Demo` is really just from LG Chat, right?

Comment (TheSonOfThomp) [reply .storybook/preview.ts:50]: Yeah, the `preview.ts` parameters is statically analyzed, so it needs to be repo-specific

**Pull request merged**